src/librustc_target/spec/armv7r_none_eabihf.rs

```diff
@@ -0,0 +1,40 @@
+// Copyright 2018 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+// Targets the Little-endian Cortex-R4F/R5F processor (ARMv7-R)
+
+use std::default::Default;
+use spec::{LinkerFlavor, PanicStrategy, Target, TargetOptions, TargetResult};
+
+pub fn target() -> TargetResult {
+    Ok(Target {
+        llvm_target: "armv7r-none-eabihf".to_string(),
+        target_endian: "little".to_string(),
+        target_pointer_width: "32".to_string(),
+        target_c_int_width: "32".to_string(),
+        data_layout: "e-m:e-p:32:32-i64:64-v128:64:128-a:0:32-n32-S64".to_string(),
+        arch: "arm".to_string(),
+        target_os: "none".to_string(),
+        target_env: "".to_string(),
+        target_vendor: "".to_string(),
+        linker_flavor: LinkerFlavor::Gcc,
+
+        options: TargetOptions {
+            executables: true,
+            relocation_model: "static".to_string(),
+            panic_strategy: PanicStrategy::Abort,
+            features: "+vfp3,+d16,+fp-only-sp".to_string(),
```

Comment:
@paoloteti I was working on a PR for this already (although I'm more than happy to defer to yours!). The set of features I pulled from the LLVM `.ts` are a bit different than yours (see [this gist](https://gist.github.com/kenkeiter/c2add8253ba10791566a2d13c42e6442)) – any thoughts?

Reply:
There are redundant features in your spec: for example v7 is superfluous because part of llvm_target (the v7 in the triple name), the same for v7r, also thumb2 is already part of +v7. +dsp and +hwdiv are superfluous too.

BTW the idea is to provide only a set of minimal features for a give target and to override/extend features at command line.

I found an RM4x board in a box in my lab. I did some test and I'm confident enough.

Reply:
Ahh, okay – wonderful. I had a feeling that you were trying to maintain a minimum subset to address the largest number of MCUs, but I wasn't aware you could apply additional features using the CLI. In that case, the set you specified here obviously looks sufficient.